src/dmd/root/speller.d

```diff
@@ -43,59 +43,53 @@ private bool combineSpellerResult(T)(ref T p, ref int cost, T np, int ncost)
     return false;
 }
 
-private auto spellerY(alias dg)(const(char)* seed, size_t seedlen, size_t index, ref int cost)
+private auto spellerY(alias dg)(const(char)[] seed, size_t index, ref int cost)
 {
-    if (!seedlen)
+    if (!seed.length)
         return null;
-    assert(seed[seedlen] == 0);
     char[30] tmp;
-    char* buf;
-    if (seedlen <= tmp.sizeof - 2)
-        buf = tmp.ptr;
+    char[] buf;
+    if (seed.length <= tmp.sizeof - 1)
+        buf = tmp;
     else
     {
-        buf = cast(char*)alloca(seedlen + 2); // leave space for extra char
-        if (!buf)
-            return null; // no matches
+        buf = new char[](seed.length + 1); // leave space for extra char
     }
     buf[0 .. index] = seed[0 .. index];
     cost = int.max;
     searchFunctionType!dg p = null;
     int ncost;
     /* Delete at seed[index] */
-    if (index < seedlen)
+    if (index < seed.length)
     {
-        buf[index .. seedlen] = seed[index + 1 .. seedlen + 1];
-        assert(buf[seedlen - 1] == 0);
-        auto np = dg(buf[0 .. seedlen - 1], ncost);
+        buf[index .. seed.length - 1] = seed[index + 1 .. seed.length];
+        auto np = dg(buf[0 .. seed.length - 1], ncost);
         if (combineSpellerResult(p, cost, np, ncost))
             return p;
     }
     /* Substitutions */
-    if (index < seedlen)
+    if (index < seed.length)
     {
-        buf[0 .. seedlen + 1] = seed[0 .. seedlen + 1];
+        buf[0 .. seed.length] = seed[0 .. seed.length];
         foreach (s; idchars)
         {
             buf[index] = s;
             //printf("sub buf = '%s'\n", buf);
-            auto np = dg(buf[0 .. seedlen], ncost);
+            auto np = dg(buf[0 .. seed.length], ncost);
             if (combineSpellerResult(p, cost, np, ncost))
                 return p;
         }
-        assert(buf[seedlen] == 0);
     }
     /* Insertions */
-    buf[index + 1 .. seedlen + 2] = seed[index .. seedlen + 1];
+    buf[index + 1 .. seed.length + 1] = seed[index .. seed.length];
```

Comment:
RHS == `seed`

Reply:
Not here, it's not starting at 0. But I switched it to $ instead of seed.length anyway.